### PR TITLE
PWN-3272 - Fix error when trying to buy USDC with Moonpay

### PR DESCRIPTION
**Pull request by gslevinkov**
## Jira Ticket (Mandatory)

https://p2pvalidator.atlassian.net/browse/PWN-3272

## Description of Work (Mandatory)

- fix error when trying to buy USDC
- fix _SOL suffix with USDC when invalid amount is entered
- **(optional)** cleanup according to styleguide, no logic is touched, only extraction

### CODE REVIEW

File: app/src/main/java/org/p2p/wallet/home/HomeModule.kt [hunk 1]
```diff
@@ -78,15 +77,37 @@ object HomeModule : InjectionModule {
...
+                qrCodeInteractor = get(),
+                usernameInteractor = get(),
+                tokenKeyProvider = get(),
+                receiveAnalytics = get(),
+                context = get()
+            )
+        }
+        factory<ReceiveNetworkTypeContract.Presenter> { (type: NetworkType) ->
+            ReceiveNetworkTypePresenter(
+                renBtcInteractor = get(),
```
Comment (gslevinkov) [inline app/src/main/java/org/p2p/wallet/home/HomeModule.kt:100]: i've anticipated Kostya's "named args plz"

File: app/src/main/java/org/p2p/wallet/moonpay/ui/BuySolanaPresenter.kt [hunk 2]
```diff
@@ -110,100 +112,130 @@ class BuySolanaPresenter(
...
+        val enteredAmountHigherThanMin = enteredAmountBigDecimal >= loadedBuyCurrency.minAmount
+
+        if (enteredAmountHigherThanMin && enteredAmountLowerThanMax) {
+            handleEnteredAmountValid(buyCurrencyInfo)
+        } else {
+            handleEnteredAmountInvalid(loadedBuyCurrency)
+        }
+    }
+
+    private fun handleEnteredAmountValid(buyCurrencyInfo: BuyCurrency) {
```
Comment (gslevinkov) [inline app/src/main/java/org/p2p/wallet/moonpay/ui/BuySolanaPresenter.kt:188]: maybe "show" naming would be better here

File: app/src/main/java/org/p2p/wallet/moonpay/ui/BuySolanaPresenter.kt [hunk 3]
```diff
@@ -216,12 +248,8 @@ class BuySolanaPresenter(
...
 
-    private fun Token.getTokenSymbolForMoonPay(): String {
-        val tokenLowercase = token.tokenSymbol.lowercase()
-        return if (token.isUSDC) {
-            "${tokenLowercase}_${SOL_SYMBOL.lowercase()}"
-        } else {
-            tokenLowercase
-        }
+    override fun detach() {
+        calculationJob?.cancel()
```
Comment (gslevinkov) [inline app/src/main/java/org/p2p/wallet/moonpay/ui/BuySolanaPresenter.kt:252]: should I cancel a custom job if we detach from the fragment?

**Review by kabramovp2p — APPROVED** [review]

**Comment by kabramovp2p** [comment]
LGTM

File: app/src/main/java/org/p2p/wallet/moonpay/repository/MoonpayApiMapper.kt [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+        val errorMessage = error.getDirectMessage() ?: error.localizedMessage
+        return errorMessage.removeUnderscoreSolIfUsdc()
+    }
+
+    /**
+     * USDC from moonpay comes with _SOL suffix, so remove it
+     * @return USDC without _SOL suffix
+     */
+    private fun String.removeUnderscoreSolIfUsdc(): String {
+        return if (contains(USDC_SYMBOL, ignoreCase = true)) {
```
Comment (gslevinkov) [inline app/src/main/java/org/p2p/wallet/moonpay/repository/MoonpayApiMapper.kt:42]: @kabramovp2p

**Review by kabramovp2p — APPROVED** [review]

**Review by 1sakov — APPROVED** [review]

**Pull request merged**